### DISCUSSION TITLE
Variable final time point of integrator

### DISCUSSION THREAD

**Discussion by moesphere** [casadi/casadi#3249] [Q&A] [answered]
Is it possible to have the final time point of an integrator as variable, i.e., to add the final time point to the optimization variables in the resulting NLP?

```
import casadi as cas
t = cas.MX.sym('t')
tf = cas.MX.sym('tf') # final integration time point
x = cas.MX.sym('x')
ode_system = dict(t=t, x=[x], ode=[-x])
F = cas.integrator('I', 'cvodes', ode_system, 0, tf)
```

**Comment by jaeandersson** [comment, marked as answer]
Yes, but the way you do it is to make a time transformation with the duration as a parameter:
```python
import casadi as cas
t = cas.MX.sym('t')
tf = cas.MX.sym('tf') # final integration time point
x = cas.MX.sym('x')
ode_system = dict(t=t, x=x, p = tf, ode=-tf*x)
F = cas.integrator('I', 'cvodes', ode_system, 0, 1)
```

**Reply by moesphere** [reply]
Thank you for the fast reply!

**Reply by brunoml5** [reply]
Hi, Joel!

I noticed that you did not expand the ode with an equation for time, such as `dt = tf`. Is it because you included `t=t` in the options for the `integrator`?

The documentation is not clear on how to use the field `t`.

Thanks
